### ISSUE TITLE
False positives when used with typescript-eslint 4.9.0

### ISSUE THREAD

**Issue by uturnr** [sweepline/eslint-plugin-unused-imports#21]
With the following versions in `package.json`:
```
    "@typescript-eslint/eslint-plugin": "4.9.0",
    "@typescript-eslint/parser": "4.9.0",
```

I get a warning that is reminiscent of an issue fixed in typescript-eslint 4.1.0 (https://github.com/typescript-eslint/typescript-eslint/pull/2496)

With this in a .d.ts file:

```
declare module 'test' {
  interface Test {
    test: 'test'
  }
}
```

I get the warning: `'Test' is defined but never used. (eslintunused-imports/no-unused-vars-ts)`

There were adjustments to `no-unused-vars` in typescript-eslint 4.9.0 that may have broken something. (https://github.com/typescript-eslint/typescript-eslint/pull/2768)

Locking my typescript-eslint version to 4.5.0 (consistent with this library) resolves the issue, but thought I would note what I've experienced.

**Comment by uturnr** [comment]
Whoops! For some reason I though that library was at fault - I just tested and am getting the same error from typescript-eslint itself (of course!) 🤦
